app/src/main/java/org/mozilla/fenix/settings/SyncPreferenceView.kt

```diff
@@ -74,17 +70,27 @@ class SyncPreferenceView(
      */
     private fun updateSyncPreferenceStatus() {
         syncPreference.apply {
-            val syncEnginesStatus = SyncEnginesStorage(context).getStatus()
-            val loginsSyncStatus = syncEnginesStatus.getOrElse(syncEngine) { false }
+            widgetVisible = true
 
-            summary = context.getString(
-                if (loginsSyncStatus) R.string.preferences_passwords_sync_logins_on
-                else R.string.preferences_passwords_sync_logins_off
-            )
+            when (key) {
+                // Credit Cards
+                context.getString(R.string.pref_key_credit_cards_sync_cards_across_devices) -> {
+                    title = context.getString(R.string.preferences_credit_cards_sync_cards)
 
-            setOnPreferenceClickListener {
-                onSyncStatusClicked()
-                true
+                    // Sync cards setOnPreferenceChangeListener should go here
+                }
+
+                // Logins
+                context.getString(R.string.pref_key_sync_logins) -> {
+                    title = context.getString(R.string.preferences_passwords_sync_logins)
+                    val syncEnginesStatus = SyncEnginesStorage(context).getStatus()
+                    val loginsSyncStatus = syncEnginesStatus.getOrElse(syncEngine) { false }
+                    isChecked = loginsSyncStatus
+                    setOnPreferenceChangeListener { _, newValue ->
+                        SyncEnginesStorage(context).setStatus(syncEngine, newValue as Boolean)
+                        true
+                    }
```

Comment:
Let's make use of some new lines here to make things more readable

```suggestion
                    title = context.getString(R.string.preferences_passwords_sync_logins)

                    val syncEnginesStatus = SyncEnginesStorage(context).getStatus()
                    val loginsSyncStatus = syncEnginesStatus.getOrElse(syncEngine) { false }

                    isChecked = loginsSyncStatus

                    setOnPreferenceChangeListener { _, newValue ->
                        SyncEnginesStorage(context).setStatus(syncEngine, newValue as Boolean)
                        true
                    }
```